### PR TITLE
Make collisionMoveSimple time overflow message show up at max once pe…

### LINKED ISSUE #3308: Terminal is Being Spammed by Warning collisionMoveSimple Error.
Minetest Engine and Game:   Compiled from 2015_10_27 GitHub but not sure which particular commit from that day.

I've noticed my terminal being spammed by the same error message, many times per second. As yet I'm not able to figure out what it is that I'm doing to trigger it.

```
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:35:44: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 19:37:31: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
[...]
2015-10-29 19:50:52: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:05:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:05:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:05:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:05:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:05:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:05:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
2015-10-29 20:19:20: WARNING[Main]: collisionMoveSimple: maximum step interval exceeded, lost movement details!
```

### PR DESCRIPTION
**Pull request by sapier**
…r step

fixes #3308

### CODE REVIEW

**Comment by est31** [comment]
I think that removing the error log is much better than adding a static boolean, which is very bad style.

:-1:

**Comment by sapier** [comment]
est31 this is a serious issue imho we shouldn't silently ignore it because if this happens something is broken for sure. We just don't know how badly it is broken.
AND this error message was in there for quite some time without causing any issue so something added to a default mod or core must cause major delays withing minetests main thread!

**Comment by est31** [comment]
> AND this error message was in there for quite some time without causing any issue so something added to a default mod or core must cause major delays withing minetests main thread!

It was probably an infostream message that was prepended with "WARNING". The logging cleanup then made it warningstream.

**Comment by sapier** [comment]
Ok made it a information again, yet spaming logefile is still not desired

File: src/collision.cpp [hunk 1]
```diff
@@ -199,18 +199,24 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
...
 
 	/*
 		Calculate new velocity
 	*/
-	if( dtime > 0.5 ) {
-		warningstream<<"collisionMoveSimple: maximum step interval exceeded, lost movement details!"<<std::endl;
+	if ( dtime > 0.5 ) {
+		if (!time_notification_done) {
+			time_notification_done = true;
+			infostream <<"collisionMoveSimple: maximum step interval exceeded, lost movement details!"<<std::endl;
```
Comment (est31) [inline src/collision.cpp]: space before and after <<

File: src/collision.cpp [hunk 2]
```diff
@@ -199,18 +199,24 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
...
+	ScopeProfiler sp(g_profiler, "collisionMoveSimple avg", SPT_AVG);
 
 	collisionMoveResult result;
 
 	/*
 		Calculate new velocity
 	*/
-	if( dtime > 0.5 ) {
-		warningstream<<"collisionMoveSimple: maximum step interval exceeded, lost movement details!"<<std::endl;
+	if ( dtime > 0.5 ) {
```
Comment (est31) [inline src/collision.cpp]: No space after ( and before ):

so like: `if ( dtime > 0.5 ) {`.

**Comment by est31** [comment]
If the style issues are fixed, and the stuff is squashed, I :+1: .

File: src/collision.cpp (hunk 2 above)
Comment (HybridDog) [inline src/collision.cpp]: it already looks this way

Comment (est31) [inline src/collision.cpp]: Ah yeah, sorry, it should look like: `if (dtime > 0.5) {`

**Pull request merged**